pom.xml

```diff
@@ -11,7 +11,23 @@
     <properties>
         <maven.compiler.release>21</maven.compiler.release>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <junit.jupiter.version>5.10.1</junit.jupiter.version>
+        <assertj.core.version>3.25.2</assertj.core.version>
     </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>${junit.jupiter.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>${assertj.core.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```

Comment:
Either merge your branch with main first or just copy pom file from the main. Otherwise it's going to conflict.

Reply:
You are absolutely right! We will as soon as possible update our pom.xml file so that it reflects main's pom.xml.
You might now if there is any difference of having the versions in 'properties' vs 'dependencies'. In main's pom.xml the versions are placed in dependencies, whereas we replicated our previous projects pom.xml where versions were placed in properties. 🤔

Reply:
While I prefer the approach with the `<properties>` for managing Maven dependencies myself, in our current scenario, there isn't much difference for several reasons.

1. Limited Dependencies: Since we don't have many dependencies, managing versions directly within each `<dependency>` doesn't pose a significant challenge. With fewer dependencies, maintaining version information within the `<dependency>`  doesn't clutter the pom.xml excessively.
2. Single Project Structure: Thunder isn't structured as a multi-module project. In a multi-module setup, utilizing a [parent POM](https://www.baeldung.com/maven-multi-module) to manage common configurations and dependency versions across modules would be beneficial. However, in our single project structure, the need for such centralized version management is minimized.
3.  Short-term project scope: Given that this is a two-week project, the likelihood of significant expansion in terms of dependencies seems low based on past project trends. 

Maybe @kappsegla can inform us more about the differences between them. But that's how I understand the distinctions between them.

Considering these factors, we have a couple of options:

1. Using `<properties>`: If you decide to go with the `<properties>` approach, you can first merge the main branch into your working branch to ensure that you have all dependencies from the main. Then, you can rewrite all versions to  `<properties>`. While this might result in conflicts during the pull request, you can address them directly within the PR interface.
![image](https://github.com/fungover/thunder/assets/42122071/d20f6d49-8a89-404a-8384-1be9b6fd2e3c)
2. Sticking with `<dependency>`: Alternatively, if the you guys prefer to stick with specifying versions directly in `<dependency>` , you can simply merge main branch with this bracn without any extensive version refactoring. (although you stil need to resolve conflict but it will be 1 conflict less then previos approach)

Reply:
One more thing to consider: currently, we have ongoing PRs #36 and #33. If we opt for the first variant (using `<properties>`), these PRs will also encounter conflicts because the POM has been modified and they branched off the old main. So, just be aware of this.

Reply:
Wow! Many thanks for the info and the heads up. We'll take a look.

Reply:
Good explanation @Ebyrdeu. I concur in your analyzing about not introducing conflicts with other PR and stay with using version numbers in the dependency tag.

Reply:
> Either merge your branch with main first or just copy pom file from the main. Otherwise it's going to conflict.

Went with merging, thanks for the help!